lib/ddtrace/contrib/rack/request_queue.rb

```diff
@@ -12,14 +12,14 @@ def get_request_start(env, now = Time.now.utc)
           header = env[REQUEST_START] || env[QUEUE_START]
           return unless header
 
-          # nginx header is in the format "t=1512379167.574"
-          # TODO: this should be generic enough to work with any
-          # frontend web server or load balancer
-          time_string = header.split('t=')[1]
+          # nginx header is milliseconds in the format "t=1512379167.574"
+          # apache header is microseconds in the format "t=1570633834463123"
+          # heorku header is milliseconds in the format "1570634024294"
+          time_string = header.to_s.gsub(/t=|\./, '')
           return if time_string.nil?
 
           # Return nil if the time is clearly invalid
-          time_value = time_string.to_f
+          time_value = "#{time_string[0, 10]}.#{time_string[10, 13]}".to_f
```

Comment:
This is quite clever!
One thing I suggest is reducing the number of fractional digits to 6, as that's the highest precision we currently expect.
But I'm curious, why you chose 13 fractional digits?